tests/Factory/ClientFactory.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+use GuzzleHttp\Client as GuzzleClient;
+use OpenAI\Client;
+use OpenAI\Factory\ClientFactory;
+use OpenAI\Transporters\HttpTransporter;
+use OpenAI\ValueObjects\ApiKey;
+use OpenAI\ValueObjects\Transporter\BaseUri;
+use OpenAI\ValueObjects\Transporter\Headers;
+
+it('may create a client', function () {
+    $openAI = (new ClientFactory('foo'))
+        ->make();
+
+    $expected = new Client(
+        new HttpTransporter(
+            new GuzzleClient(),
+            BaseUri::from('api.openai.com/v1'),
+            Headers::withAuthorization(ApiKey::from('foo'))
+        )
+    );
+
+    expect($openAI)->toEqual($expected);
+});
+
+it('may create a client with base URI', function () {
+    $openAI = (new ClientFactory('foo'))
+        ->withBaseUri('http://api.custom-domain.com/v1')
+        ->make();
+
+    $expected = new Client(
+        new HttpTransporter(
+            new GuzzleClient(),
+            BaseUri::from('http://api.custom-domain.com/v1'),
+            Headers::withAuthorization(ApiKey::from('foo'))
+        )
+    );
+
+    expect($openAI)->toEqual($expected);
```

Comment:
What exactly this `toEqual` is testing? You may have to use reflection - to see if the client was created with the correct information instead.

Reply:
`toEqual` tests, if the correct base URI / organization was injected into the `HttpTransporter`.
The test fails, when that's not the case (e.g. when `withBaseUri` does nothing).

See also https://3v4l.org/ZobO3#v8.1.13 -> Comparing objects with `==` compares the primitive values in the object (in contrast to `===`).

Or what do you mean?

Reply:
lol, TIL.